File: 0260-single-number-iii/0260-single-number-iii.py
class Solution:
    def singleNumber(self, nums: List[int]) -> List[int]:
        res = []
        i = 0
        nums = sorted(nums)
        
        while i < len(nums):
            if i < len(nums) - 1 and nums[i] == nums[i + 1]:
                i += 2
            else:
                res.append(nums[i])
                i += 1
        return res

File: 0260-single-number-iii/0260-single-number-iii.java
class Solution {
    public int[] singleNumber(int[] nums) {
        ArrayList<Integer> list = new ArrayList<>();
        int i = 0;
        Arrays.sort(nums);
        
        while (i < nums.length) {
            if(i < nums.length - 1 && nums[i] == nums[i + 1]) {
                i += 2;
            }
            else {
                list.add(nums[i]);
                i += 1;
            }
        }
        
        int[] arr = new int[list.size()];
        
        for(int j = 0; j < list.size(); j++) {
            arr[j] = list.get(j);
        }        
        return arr;
    }
}